src/main/java/net/imglib2/img/basictypeaccess/nio/DoubleBufferAccess.java

```diff
@@ -0,0 +1,238 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imglib2.img.basictypeaccess.nio;
+
+import java.nio.ByteBuffer;
+import java.nio.DoubleBuffer;
+
+import net.imglib2.img.basictypeaccess.array.AbstractDoubleArray;
+import net.imglib2.img.basictypeaccess.volatiles.VolatileDoubleAccess;
+
+/**
+ * Access for {@link DoubleBuffer}
+ *
+ * @author Mark Kittisopikul
+ */
+public class DoubleBufferAccess extends AbstractBufferAccess< DoubleBufferAccess, DoubleBuffer > implements VolatileDoubleAccess
+{
+
+	/**
+	 * Automatically generated
+	 */
+	private static final long serialVersionUID = -7265085228179236189L;
+
+	private static final int NUM_BYTES_PER_ENTITY = Double.BYTES;
+
+	public DoubleBufferAccess( final DoubleBuffer buffer, final boolean isValid )
+	{
+		super( buffer, isValid );
+	}
+
+	public DoubleBufferAccess( final int numEntities, final boolean isValid )
+	{
+		super( DoubleBuffer.allocate( numEntities ), isValid );
+	}
+
+	public DoubleBufferAccess( final ByteBuffer buffer, final boolean isValid )
+	{
+		super( buffer.asDoubleBuffer(), isValid );
+	}
+
+	// Convenience constructors
+
+	public DoubleBufferAccess( final DoubleBuffer buffer )
+	{
+		this( buffer, DEFAULT_IS_VALID );
+	}
+
+	public DoubleBufferAccess( final int numEntities )
+	{
+		this( numEntities, DEFAULT_IS_VALID );
+	}
+
+	public DoubleBufferAccess( final ByteBuffer buffer )
+	{
+		this( buffer, DEFAULT_IS_VALID );
+	}
+
+	public DoubleBufferAccess()
+	{
+		this( ( DoubleBuffer ) null, false );
+	}
+
+	/*
+	 * DoubleAccess methods
+	 */
+
+	@Override
+	public double getValue( final int index )
+	{
+		return buffer.get( index );
+	}
+
+	@Override
+	public void setValue( final int index, final double value )
+	{
+		buffer.put( index, value );
+	}
+
+	/*
+	 * AbstractBufferAccess methods
+	 */
+
+	@Override
+	public int getNumBytesPerEntity()
+	{
+		return NUM_BYTES_PER_ENTITY;
+	}
+
+	@Override
+	public DoubleBufferAccess newInstance( final ByteBuffer buffer, final boolean isValid )
+	{
+		return fromByteBuffer( buffer, isValid );
+	}
+
+	@Override
+	public DoubleBufferAccess newInstance( final DoubleBuffer buffer, final boolean isValid )
+	{
+		return new DoubleBufferAccess( buffer, isValid );
+	}
+
+	@Override
+	protected DoubleBuffer duplicateBuffer( final DoubleBuffer buffer )
+	{
+		return buffer.duplicate();
+	}
+
+	/**
+	 * Override abstract implementation to allow for longer non-direct Buffers
+	 * since ByteBuffer is restricted to Integer.MAX_VALUE entities.
+	 */
+	@Override
+	protected DoubleBufferAccess allocate( final int numEntities, final boolean isDirect, final boolean isValid )
+	{
+		if ( isDirect )
+			return super.allocate( numEntities, isDirect, isValid );
+		else
+			return new DoubleBufferAccess( numEntities, isValid );
+	}
```

Comment:
@hanslovsky I ended overloading allocate like this before I saw your comment.

Reply:
That works for me, too.